installer/templates/phx_ecto/data_case.ex

```diff
@@ -45,8 +45,8 @@ defmodule <%= app_module %>.DataCase do
   """
   def errors_on(changeset) do
     Ecto.Changeset.traverse_errors(changeset, fn {message, opts} ->
-      Enum.reduce(opts, message, fn {key, value}, acc ->
-        String.replace(acc, "%{#{key}}", to_string(value))
+      Regex.replace(~r"%{(\w+)}", message, fn _, key ->
+        Map.get(opts, String.to_atom(key), key) |> to_string()
```

Comment:
I think the use of `String.to_atom/1` here is acceptable since they string is a "trusted" value.

If we're not comfortable with this, I can first transform the opts using the logic in @wojtekmach 's PR comment so that we don't need to use `String.to_atom/1`. I personally just liked the conciseness of the code.

Reply:
For a test helper function String.to_atom is imo acceptable 👍